src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.FindLiteralsServerCallback.cs

```diff
@@ -28,11 +30,29 @@ public ValueTask AddItemsAsync(int count)
             public ValueTask ItemCompletedAsync()
                 => _progress.ProgressTracker.ItemCompletedAsync();
 
-            public async ValueTask OnReferenceFoundAsync(DocumentId documentId, TextSpan span)
+            public async ValueTask OnLiteralReferenceFoundAsync(DocumentId documentId, TextSpan span)
             {
                 var document = _solution.GetDocument(documentId);
                 await _progress.OnReferenceFoundAsync(document, span).ConfigureAwait(false);
             }
+
+            public ValueTask OnCompletedAsync()
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask OnDefinitionFoundAsync(SerializableSymbolAndProjectId definition)
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask OnFindInDocumentCompletedAsync(DocumentId documentId)
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask OnFindInDocumentStartedAsync(DocumentId documentId)
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask OnReferenceFoundAsync(SerializableSymbolAndProjectId definition, SerializableReferenceLocation reference)
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask OnStartedAsync()
+                => throw ExceptionUtilities.Unreachable;
```

Comment:
will admit i'm a bit confused why we implement the interface only to throw on every new entry in it.

Reply:
IServiceBroker expects a service interface to have at most one callback interface. The symbol finder service used two different callback interfaces in different RPC method calls. This interface thus needs to implement a union of the methods of those two callback interfaces but each subset is only implemented by method-specific instance on the client.